packages/node_modules/@ciscospark/widget-space/src/enhancers/setup.js

```diff
@@ -106,6 +109,27 @@ function setup(props) {
     && sparkState.get('registered')
     && !sparkState.get('hasError')
   ) {
+    // Check if we need to reload space
+    if (widgetStatus.shouldReloadWidget && prevProps && !prevProps.widgetStatus.shouldReloadWidget) {
+      props.reloadWidget();
+      props.resetErrors();
+    }
+
+    // Check for destination Change
+    if (destination && prevProps.destination) {
+      const calculatedDestination = getDestinationFromProps(props);
+      const calculatedPreviousDestination = getDestinationFromProps(prevProps);
+      if (
+        calculatedPreviousDestination.id !== calculatedDestination.id
+        || calculatedPreviousDestination.type !== calculatedDestination.type
+      ) {
```

Comment:
Should we also take into account if someone passes an email, and then passes in the `spaceId` of the same space? Instead of comparing current to previous props could we, instead, compare current props, to what we have in the space/conversation store?
That would also make our migration to a centralized spaces store a lot easier.

Reply:
Is this an over-optimization? Seems like a VERY rare case to code for. I'm not even sure how to go able getting started on implementing it.

Reply:
Not really an over-optimization, more just looking forward to how this will be used when we're not using the single conversation store, but the spaces store. We should minimize server calls as much as possible to provide a responsive UI to the end user. If we have the data already, we shouldn't go and refetch it.

This accomplishes what we need for now.  We'll just eventually need check what's in the store and not only the previous prop to detect the need to fetch or to load from store.

Reply:
@bzang I've been trying to figure out how to implement this in the space widget without relying on conversation service, but we still don't have a way to get a space via email. Any ideas?

Reply:
One approach might be to get the "other" user in a convo (I think we have that in a selector somewhere). If you check if the convo is `one_on_one` and the other user's email matches, then that should be a match for the space to reuse.

Maybe create a single function that takes the email and conversation and determines that. We can then easily migrate it into the spaces module alter.

Reply:
Ok, I see, let me give it a shot.